### PR TITLE
fix: relative url error related to console errors

### PR DESCRIPTION
**Pull request by sowmyadev-kota**
#### Description of changes:
Fix `Console Errors`: https://github.com/aws-amplify/docs/actions/runs/6580015445/job/17877101477?pr=5966

Staging site: https://internal-links.d1ywzrxfkb9wgg.amplifyapp.com/

`Errors Reasons`
urls like this will produce a double slash /lib/datastore/setup-auth-rules/ this also includes urls with a slash and hash /lib/datastore/setup-auth-rules/#something an example of this can be found on page http://localhost:3000/lib/datastore/sync/q/platform/android/ and the link for this is in src/fragments/lib/datastore/android/sync/50-selectiveSync.mdx

This will resolve the error

Invalid href '/lib-v1/datastore/setup-auth-rules//q/platform/android#configure-multiple-authorization-types' passed to next/router in page: '/lib-v1/graphqlapi/authz/q/platform/[platform]'. Repeated forward-slashes (//) or backslashes \ are not valid in the href. found on http://localhost:3000/lib-v1/graphqlapi/authz/q/platform/android/
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by reesscot — CHANGES_REQUESTED** [review]
Blocking for now since the staging build is failing

File: src/components/InternalLink/index.tsx [hunk 1]
```diff
@@ -28,9 +28,17 @@ export default function InternalLink({ href, children }) {
         const filterKey = filterKeys[filterKind];
         if (href.includes('#')) {
           const hrefParts = href.split('#');
-          href = `${hrefParts[0]}/q/${filterKind}/${filterKey}#${hrefParts[1]}`;
+          if (hrefParts[0].endsWith('/')) {
+            href = `${hrefParts[0]}q/${filterKind}/${filterKey}#${hrefParts[1]}`;
+          } else {
+            href = `${hrefParts[0]}/q/${filterKind}/${filterKey}/${hrefParts[1]}`;
```
Comment (jacoblogan) [inline src/components/InternalLink/index.tsx:34]: missing # on this line `${hrefParts[0]}/q/${filterKind}/${filterKey}#${hrefParts[1]}`

Comment (sowmyadev-kota) [reply src/components/InternalLink/index.tsx]: Thank you!!

**Comment by jacoblogan** [comment]
Ran the console error script locally against this change and all of the // errors have been resolved

**Review by jacoblogan — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]
LGTM, thanks @sowmyadev-kota

**Review by reesscot — APPROVED** [review]

**Pull request merged**